Controllers/HomeController.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Zoltu.Bags.Client.Controllers
+{
+	public class HomeController : Controller
+	{
+		[Route("", Name ="Home", Order = 0)]
+		[Route("app", Name = "app", Order = 0)]
+		public IActionResult App()

```

Comment:
There is one route missing here, the `[Route("app/{*path}"]` route.  It should have an `Order` that is lower than the `"app"` route so it is evaluated only if `app` doesn't match.

This will require adding a parameter to the `App()` method for this with a default value of null:
```
public IActionResult App(String path = null)
```